crates/subspace-fraud-proof/src/invalid_state_transition_proof.rs

```diff
@@ -240,6 +257,13 @@ pub trait VerifyPrePostStateRoot {
         &self,
         invalid_state_transition_proof: &InvalidStateTransitionProof,
     ) -> Result<(), VerificationError>;
+
+    /// Returns the hash of primary block at height `domain_block_number`.
+    fn primary_hash(
```

Comment:
Adding the func to this trait seems weird, Maybe a different trait would be nicer

Reply:
Agree. I left it as is intentionally for simplicity as we'll probably rename the trait `VerifyPrePostStateRoot` to be something more appropriate (I don't yet have a good idea about this :P) when refactoring `pre_state_root`, and then another trait is unnecessary.